external/loaders/loaders/mappers/_hybrid.py

```diff
@@ -1,117 +1,85 @@
+    approach: str,
+    fine_url: str,
+    include_temperature_nudging: bool = False,
+    additional_dataset_urls: Sequence[str] = None,
+    use_fine_res_state: bool = True,
+    use_fine_res_fluxes: bool = False,
 ) -> GeoMapper:
     """
-    Open the fine resolution nudging_hybrid mapper
+    Open the fine-res mapper using several configuration options
 
     Args:
+        approach: one of a set of available approaches.
         fine_url: url where coarsened fine resolution data is stored
-        nudge_url: url to nudging data to be used as a residual
         include_temperature_nudging: whether to include fine-res nudging in Q1
+        additional_dataset_urls: sequence of urls to zarrs containing additional
+            data to be merged into the resulting mapper dataset, e.g., ML input
+            features, the dynamics nudging tendencies, and the dynamics differences
+            as required by the above approaches
+        use_fine_res_state: set standard name state variables to point to the fine-res
+            data. Set to True if wanting to use fine-res state as ML inputs in training.
+        use_fine_res_fluxes: set standard name surface and TOA flux diagnostic variables
+            to point to the fine-res data. Set of True if wanting to use fine-res fluxes
+            as ML inputs in training.
 
     Returns:
         a mapper
     """
-    return XarrayMapper(
-        _open_fine_resolution_nudging_hybrid_dataset(
-            fine_url=fine_url,
-            nudge_url=nudge_url,
-            include_temperature_nudging=include_temperature_nudging,
-        )
+
+    approach_enum = Approach[approach]
+
+    merged: FineResBudget = _open_merged_dataset(
+        fine_url=fine_url,
+        additional_dataset_urls=additional_dataset_urls,
+        use_fine_res_state=use_fine_res_state,
+        use_fine_res_fluxes=use_fine_res_fluxes,
     )
+    budget = compute_budget(merged, approach_enum, include_temperature_nudging)
+    hybrid_budget = compute_hybrid_budget(budget)
+
+    return XarrayMapper(hybrid_budget)
 
 
 @mapper_functions.register
 def open_precomputed_fine_resolution_nudging_hybrid(
-    fine_url: str, nudge_url: str,
+    fine_url: str, additional_dataset_urls: str = None
 ) -> GeoMapper:
     """
     Open the fine resolution nudging hybrid mapper with precomputed fine-res data
 
     Args:
         fine_url: url where coarsened fine resolution data is stored, must include
-            precomputed Q1 and Q2
-        nudge_url: url to nudging data to be used as a residual
+            precomputed Q1, Q2, dQ1 and dQ2
+        additional_dataset_urls: sequence of urls which to zarrs containing additional
+            data to be merged into the resulting mapper dataset
 
     Returns:
         a mapper
     """
     return XarrayMapper(
-        _open_precomputed_fine_resolution_nudging_hybrid_dataset(
-            fine_url=fine_url, nudge_url=nudge_url
+        _open_merged_dataset(
```

Comment:
I think for the precomputed hybrid case, the assumption was that only the fine portion is precomputed, and that we still do the "hybrid" computations on the fly. I'd presumably retain that.

Reply:
This would also work well if the precomputed hybrid and fine res openers were the same. Would just need a flag on whether to call `compute_hybrid_budget` or not, and maybe some careful handling of the wind nudging tendencies.